compiler/type-hint.cpp

```diff
@@ -337,6 +337,19 @@ const TypeHint *TypeHintShape::create(std::vector<std::pair<std::string, const T
   );
 }
 
+std::map<std::int64_t, std::string> TypeHintShape::shape_keys_storage_;
+std::mutex TypeHintShape::shape_keys_storage_lock_;
+
+void TypeHintShape::set_shape_key(std::int64_t key_hash, std::string_view key) noexcept {
+  std::lock_guard lock{shape_keys_storage_lock_};
```

Comment:
Can we move std::mutex here, as a static variable? 
`#include<mutex>` seems quite weird in type-hint.h

Reply:
Done